ui/lib/screens/projects/projects_screen.dart

```diff
@@ -20,24 +21,31 @@ class ProjectsScreen extends StatefulWidget {
 class _ProjectsScreenState extends State<ProjectsScreen> {
   late ProjectService service;
   late Future<List<Project>> projects;
+  String _filter = '';
 
   @override
   void initState() {
     super.initState();
     service = ProjectService.of(context);
-    projects = service.allProjects();
+    projects = service.findProjects(_filter);
   }
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
-        title: Text('All projects'),
+        title: NameFilter(
+          hint: 'search for Projects',
+          onChanged: (fragment) => setState(() {
+            _filter = fragment;
+            projects = service.findProjects(_filter);
+          }),
+        ),
         actions: [
           IconButton(
             icon: Icon(Icons.refresh),
             onPressed: () => setState(() {
-              projects = service.allProjects();
+              projects = service.findProjects(_filter);
```

Comment:
I don't think reload still makes a lot of sense in combination with the search bar. (I added it because there previously was no way to update the list.)

Let's remove the action button here (the packages screen also has no reload button).

PS: I think you then also don't need to track the `_filter` value anymore.